frontend/src/components/course/course.tsx

```diff
@@ -231,19 +230,21 @@ const Course: FC = () => {
         </div>
       </div>
 
-      {isUserAuthorized && (
+      {isUserAuthorized && isMentorDataStatus === DataStatus.FULFILLED && (
         <div className={styles.additional}>
-          {mentor && (
-            <MyMentor
-              mentor={mentor}
-              onMentorChange={handleChooseMentorModalToggle}
-            />
-          )}
-          {isMentor && menteesByCourseDataStatus === DataStatus.FULFILLED && (
-            <MyStudentsContainer mentees={mentees} />
-          )}
-          {isMentorChoosingEnabled && (
+          {isMentor ? (
+            menteesByCourseDataStatus === DataStatus.FULFILLED && (
+              <MyStudentsContainer mentees={mentees} />
+            )
+          ) : isMentorChoosingEnabled ? (
             <ChooseMentorButton onClick={handleChooseMentorModalToggle} />
+          ) : (
+            mentor && (
+              <MyMentor
+                mentor={mentor}
+                onMentorChange={handleChooseMentorModalToggle}
+              />
+            )
```

Comment:
Let's avoid nested ternary expressions with additional conditions inside them (split it somehow outside the render to make it readable)